### ISSUE TITLE
Proxy List target giving 502

### ISSUE THREAD

**Issue by viswatejan** [almeidapaulopt/tsdproxy#113]
**Describe the bug**
Proxy List target is giving 502 even when the target TLS validation is disabled.

**Expected behavior**
It should skip TLS validation 

**Send config**
```yaml
defaultproxyprovider: default
docker:
  local: # name of the docker target provider
    host: unix:///var/run/docker.sock # host of the docker socket or daemon
    targethostname: 192.168.0.100 # hostname or IP of docker server
    defaultproxyprovider: default # name of which proxy provider to use
files:
  home: # Name the target provider
    filename: /config/home.yaml # file with the proxy list
    defaultProxyProvider: default # (optional) default proxy provider
    defaultProxyAccessLog: true # (optional) Enable access logs
tailscale:
  providers:
    default: # name of the provider
      authkey: ---REMOVED--- # define authkey here
      controlurl: https://controlplane.tailscale.com # use this to override the default control URL
  datadir: /data/
http:
  hostname: 0.0.0.0
  port: 8080
log:
  level: trace # set logging level info, error or trace
  json: false # set to true to enable json logging
proxyaccesslog: true # set to true to enable container access log
```

**Logs**

```
4:27AM TRC cert("xyz.funny-name.ts.net"): registered ACME account. module=proxymanager tailscale=default
4:27AM TRC cert("xyz.funny-name.ts.net"): starting SetDNS call... module=proxymanager tailscale=default
4:27AM TRC wg: [v2] [SyWUl] - Receiving keepalive packet module=proxymanager tailscale=default
4:27AM TRC cert("xyz.funny-name.ts.net"): did SetDNS module=proxymanager tailscale=default
4:27AM TRC cert("xyz.funny-name.ts.net"): requesting cert... module=proxymanager tailscale=default
4:27AM TRC cert("xyz.funny-name.ts.net"): got cert module=proxymanager tailscale=default
2024/12/21 04:27:42 http: TLS handshake error from 100.69.83.27:34156: write tcp 100.127.236.45:443->100.69.83.27:34156: i/o timeout
4:27AM TRC [v1] Accept: TCP{100.69.83.27:34156 > 100.127.236.45:443} 52 tcp non-syn
 module=proxymanager tailscale=default
4:27AM ERR error host=xyz.funny-name.ts.net method=GET module=proxymanager proxyname=cam status=502 url=/
2024/12/21 04:27:43 http: proxy error: unsupported protocol scheme ""
```

**Additional context**

**Comment by almeidapaulopt** [comment]
Did you look to troubleshooting page?
 https://almeidapaulopt.github.io/tsdproxy/docs/troubleshooting/

**Comment by techathead** [comment]
I am having a similar issue, my proxy list stopped working when I upgraded to 1.2.0, although I didn't get any errors, it's like it didn't even try loading the file at all. I downgraded image tags to 1.1.2 and it worked no problem.

**Comment by theGressier** [comment]
Hello, I had an issue with 1.2.0 and it was related to incorrect typo in my config file, I was using keyword with a wrong case (URL instead of url). Did you check the config file?

**Comment by viswatejan** [comment]
My `/config/home.yaml` is as follows as documented in [Proxy List](
https://almeidapaulopt.github.io/tsdproxy/docs/files/)

```yaml
xyz:
  URL: https://192.168.0.123
  TLSValidate: false
````

I will try altering the case and check if that resolves.

**Comment by viswatejan** [comment]
> Did you look to troubleshooting page? https://almeidapaulopt.github.io/tsdproxy/docs/troubleshooting/

@almeidapaulopt, I did check the troubleshooting steps but they seem to be specific to the docker containers and not related to the Proxy List or 502.

I also checked some old reported issues with 502 and understood that this could be related to the TLS validation on the targets having the self signed certs but disabling the TLS validation didn't help to resolve this.

**Comment by almeidapaulopt** [comment]
> > Did you look to troubleshooting page? https://almeidapaulopt.github.io/tsdproxy/docs/troubleshooting/
> 
> @almeidapaulopt, I did check the troubleshooting steps but they seem to be specific to the docker containers and not related to the Proxy List or 502.
> 
> I also checked some old reported issues with 502 and understood that this could be related to the TLS validation on the targets having the self signed certs but disabling the TLS validation didn't help to resolve this.

try this:

`xyz:
  url: https://192.168.0.123
  tlsvalidate: false`

if this work. it's a case sensitive bug.

**Comment by viswatejan** [comment]
> > > Did you look to troubleshooting page? https://almeidapaulopt.github.io/tsdproxy/docs/troubleshooting/
> > 
> > 
> > @almeidapaulopt, I did check the troubleshooting steps but they seem to be specific to the docker containers and not related to the Proxy List or 502.
> > I also checked some old reported issues with 502 and understood that this could be related to the TLS validation on the targets having the self signed certs but disabling the TLS validation didn't help to resolve this.
> 
> try this:
> 
> ```
> xyz:
>   url: https://192.168.0.123
>   tlsvalidate: false
> ```
> 
> if this work. it's a case sensitive bug.

@almeidapaulopt, that worked perfectly. Thank you!

Thanks @theGressier for mentioning about the case.

**Comment by almeidapaulopt** [comment]
This is fixed with v1.3.0
